.travis.yml

```diff
@@ -24,60 +15,68 @@ stages:
 
 jobs:
   include:
-    - name: "Linters"
+    - name: 'Linters'
       stage: test
       language: python
-      python: "3.6"
+      python: '3.6'
       install:
         - pip3 install --upgrade pycodestyle
       script:
         - pycodestyle --max-line-length=99 --exclude='./python/lookout/sdk' .
 
-    - name: "Python: example integration tests"
+    - name: 'Python: example integration tests'
       stage: test
       language: python
-      python: "3.6"
+      python: '3.6'
       install:
+        - ./_tools/ci-run-bblfsh.sh
         - pip3 install -e python
       script:
         - ./_tools/install-lookout-latest.sh
-        - (python3 language-analyzer.py |& tee -a ./py-analyzer.log)&
-        - ./lookout-sdk review --log-level=debug --from 0a9d1d159d2b0064c32df8d2287b174a91390b1a --to HEAD "ipv4://localhost:2021"
-      after_failure:
-        - echo "test"
-        - cat ./py-analyzer.log
-        - cat ./lookout-install.log
+        - (python3 -u language-analyzer.py |& tee ./py-analyzer.log)&
+        - sleep 5s
+        - ./lookout-sdk review --log-level=debug --from 0a9d1d159d2b0064c32df8d2287b174a91390b1a --to HEAD "ipv4://localhost:2021" |& tee ./sdk.log
+        - grep -v -i "error" ./py-analyzer.log > /dev/null
+        - grep -v -i "error" ./sdk.log > /dev/null
 
-    - name: "Golang: example integration tests"
+    - name: 'Golang: example integration tests'
       stage: test
       language: go
-      go: "1.10"
+      go: '1.10'
       install:
         - go version
         - go get .
       script:
+        - ./_tools/ci-run-bblfsh.sh
         - ./_tools/install-lookout-latest.sh
-        - (go run language-analyzer.go |& tee -a ./go-analyzer.log)&
-        - sleep 1s
-        - ./lookout-sdk review --log-level=debug --from 0a9d1d159d2b0064c32df8d2287b174a91390b1a --to HEAD "ipv4://localhost:2020"
-      after_failure:
-        - cat ./go-analyzer.log
-        - cat ./lookout-install.log
+        - (go run language-analyzer.go |& tee ./go-analyzer.log)&
+        - sleep 5s
+        - ./lookout-sdk review --log-level=debug --from 0a9d1d159d2b0064c32df8d2287b174a91390b1a --to HEAD "ipv4://localhost:2020" |& tee ./sdk.log
+        - grep -v -i "error" ./go-analyzer.log > /dev/null
+        - grep -v -i "error" ./sdk.log > /dev/null
```

Comment:
I'm confused about this last two lines, and most probably missing something... please let me help to understand if so.

Last two commands will pipe every line from those logs, not containing `error` word into `/dev/null`, and all the lines containing `error` word will be just ignored.

But:

- errors in `language-analyzer.go` are not logged with the word `error`, but `failed` or `SIGABORT` or `abort`, example:
```
...
language-analyzer: uast.cc:268: Nodes* UastFilter(const Uast*, void*, const char*): Assertion `node' failed.
SIGABRT: abort
PC=0x7fe40990c428 m=7 sigcode=18446744073709551610
signal arrived during cgo execution
....
```
- what if in the logs it's found that word (`error`)? For example, it could appear in a file path (that is also logged)
```
time="2018-11-21T17:04:530" level=debug msg="parsing uast for file: html/partials/errors.tpl" app=lookout-sdk source="bblfsh/bblfsh.go:95"
```
Would it be detected as a false positive?
- is `grep -v -i $WORD $FILE_TO_GREP > /dev/null` making the integration test fail? Because it's always exiting with `0` (successfully); would not be better to focus in the exit code of the analyzer and `lookout-sdk` itself?

Reply:
1. I believe it depends on error. You got error from binding. Though such failure should also fail response and error will appear in sdk log.
2. We don't have such files for now. But yes, it's possible.
3. `lookout-sdk` always returns `0`. And analyzer doesn't exit at all.

I do understand it's not perfect test but in my opinion it's good enough for now. They cover 99% reasonable cases that I could think of and shouldn't cause false positives for them.

If you have any suggestions about how to test it better without creating a monster: please let me know, we will improve this checks.

Reply:
ping @dpordomingo 

any ideas on it?